### PR TITLE
Update E0318 to new format

### LINKED ISSUE #35322: E0318 needs to be updated to new format
From: src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs

Error E0318 needs a span label, a note, and an updated title, updating it from:

```
error[E0318]: cannot create default implementations for traits outside the crate they're defined in; define a new trait instead
  --> src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs:13:1
   |
13 | impl Copy for .. {}
   | ^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0318]: cannot create default implementations for traits outside the crate they're defined in
  --> src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs:13:1
   |
13 | impl Copy for .. {}
   | ^^^^^^^^^^^^^^^^^^^ `Copy` trait not defined in this crate
   |
   = note: define a new trait instead 
```

Bonus: narrow the span to focus on the foreign trait

```
error[E0318]: cannot create default implementations for traits outside the crate they're defined in
  --> src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs:13:1
   |
13 | impl Copy for .. {}
   |      ^^^^ trait not defined in this crate
   |
   = note: define a new trait instead 
```

### PR DESCRIPTION
**Pull request by acrrd**
Fixes #35322.
Part of #35233.

r? @GuillaumeGomez

### CODE REVIEW

File: src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs [hunk 1]
```diff
@@ -10,7 +10,6 @@
 
 #![feature(optin_builtin_traits)]
 
-impl Copy for .. {}
-//~^ ERROR E0318
-
+impl Copy for .. {} //~ ERROR E0318
+  //~^^^^ ERROR `Copy` trait not defined in this crate
```
Comment (GuillaumeGomez) [inline src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs]: Why so much "^^^^^"?

**Comment by acrrd** [comment]
It was to underline the same part that the error do.
I removed them.

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Comment by GuillaumeGomez** [comment]
@bors: r-

Error: 

```bash
---- [compile-fail] compile-fail\typeck-default-trait-impl-outside-crate.rs stdout ----
	
error: C:/bot/slave/auto-win-gnu-32-opt-rustbuild/build/src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs:13: expected error not found: `Copy` trait not defined in this crate

error: 0 unexpected errors found, 1 expected errors not found
status: exit code: 101
command: PATH="C:\bot\slave\..."
not found errors (from test file): [
    Error {
        line_num: 13,
        kind: Some(
            Error
        ),
        msg: "`Copy` trait not defined in this crate"
    }
]

thread '[compile-fail] compile-fail\typeck-default-trait-impl-outside-crate.rs' panicked at 'explicit panic', C:\bot\slave\auto-win-gnu-32-opt-rustbuild\build\src\tools\compiletest\src\runtest.rs:1084


failures:
    [compile-fail] compile-fail\typeck-default-trait-impl-outside-crate.rs

test result: FAILED. 2524 passed; 1 failed; 12 ignored; 0 measured
```

**Comment by acrrd** [comment]
Sorry about that, I pushed the fix.

**Comment by GuillaumeGomez** [comment]
Have you tested it? I wonder if it was waiting for a note or an error. It's not very clear...

**Comment by acrrd** [comment]
Before I put an ERROR in test, now I put NOTE and works

**Comment by GuillaumeGomez** [comment]
Fine by me then. Thanks!

@bors: r+ rollup

**Pull request merged**